src/main.rs

```diff
@@ -144,6 +165,68 @@ async fn main() -> Result<()> {
 		TriggerRepository,
 	>(None, None, None)?;
 
+	let monitor_name = matches
+		.get_one::<String>("monitorName")
+		.map(|s| s.to_string());
+	let network_slug = matches.get_one::<String>("network").map(|s| s.to_string());
+	// Check if the user wants to execute the monitor only for a specific block number or a hash
+	let block_number = matches
+		.get_one::<String>("block")
+		.map(|s| {
+			s.parse::<u64>().map_err(|e| {
+				error!("Failed to parse block number: {}", e);
+				e
+			})
+		})
+		.transpose()?;
+
+	// Check if any specific execution parameters are present
+	let monitor_test_execution =
+		monitor_name.is_some() || network_slug.is_some() || block_number.is_some();
+
+	if monitor_test_execution {
```

Comment:
Let's separate this in a new function inside `main`.
perhaps `should_test_monitor_execution` and `test_monitor_execution`